dashboard/app/controllers/courses_controller.rb

```diff
@@ -36,51 +32,17 @@ def show
       return
     end
 
-    if @unit_group.present?
-      if @unit_group.plc_course
-        authorize! :show, Plc::UserCourseEnrollment
-        user_course_enrollments = [Plc::UserCourseEnrollment.find_by(user: current_user, plc_course: @unit_group.plc_course)]
-        render 'plc/user_course_enrollments/index', locals: {user_course_enrollments: user_course_enrollments}
-        return
-      end
-
-      if @unit_group.pilot?
-        authenticate_user!
-        unless @unit_group.has_pilot_access?(current_user)
-          render :no_access
-          return
-        end
-      end
-
-      if @unit_group.in_development?
-        authenticate_user!
-        unless current_user.permission?(UserPermission::LEVELBUILDER)
-          render :no_access
-          return
-        end
-      end
-
-      # Attempt to redirect user if we think they ended up on the wrong course overview page.
-      override_redirect = VersionRedirectOverrider.override_course_redirect?(session, @unit_group)
-      if !override_redirect && redirect_unit_group = redirect_unit_group(@unit_group)
-        redirect_to "#{course_path(redirect_unit_group)}/?redirect_warning=true"
-        return
-      end
-
-      sections = current_user.try {|u| u.sections.where(hidden: false).select(:id, :name, :course_id, :script_id)}
-      @sections_with_assigned_info = sections&.map {|section| section.attributes.merge!({"isAssigned" => section[:course_id] == @unit_group.id})}
-
-      render 'show', locals: {unit_group: @unit_group, redirect_warning: params[:redirect_warning] == 'true'}
-    elsif UnitGroup.family_names.include?(params[:course_name])
-      # csp and csd are each "course families", each containing multiple "course versions".
-      # When the url of a course family is requested, redirect to a specific course version.
-      redirect_query_string = request.query_string.empty? ? '' : "?#{request.query_string}"
-      redirect_to_course = UnitGroup.latest_stable(params[:course_name])
-      redirect_to "#{course_path(redirect_to_course)}#{redirect_query_string}"
+    # Attempt to redirect user if we think they ended up on the wrong course overview page.
+    override_redirect = VersionRedirectOverrider.override_course_redirect?(session, @unit_group)
+    if !override_redirect && redirect_unit_group = redirect_unit_group(@unit_group)
+      redirect_to "#{course_path(redirect_unit_group)}/?redirect_warning=true"
       return
-    else
-      raise ActiveRecord::RecordNotFound
     end
```

Comment:
how do we know it is safe to remove lines 74-83?

Reply:
The logic covered here has moved into get_unit_group and set_unit_group.